engine/class_modules/warlock/sc_warlock_affliction.cpp

```diff
@@ -658,9 +658,6 @@ struct drain_soul_t : public affliction_spell_t
   void execute() override
   {
     dot_t* dot = get_dot( target );
-    if ( dot->is_ticking() && !p()->prolonged_decimation_trigger() )
-      p()->buffs.decimating_bolt->decrement();
```

Comment:
Do not remove this decrement here, otherwise you can chain cast Drain Soul without using any Decimating Bolt stacks.

Reply:
Thanks for the important input!
I just tested again on the SL Beta and the check ingame happens whenever the channel is aborted (regardless if finished until last tick or not). So is there maybe a better place to add this code to begin with?

Reply:
last_tick will always be called when you stop channeling, whether that's early interrupt or not. The issue is more with recasting Drain Soul while already channeling Drain Soul. The Drain Soul effect is technically never interrupted (so last_tick won't be called) but in game, it should consume a stack of the buff.

Reply:
Or, in other words, you need both the last_tick code and this code

Reply:
Okay, I think I understand that, but wouldn't that leave an unresolved issue because the buff (which would then be consumed/decremented on the first tick) contains the needed info for the spell-multiplier?

Reply:
That's why the code is checking whether the "dot" (the channel) is ticking. If you're starting a fresh channel, it won't be ticking and therefore the buff won't be consumed.

Reply:
Unless I'm misreading the PR code, this still needs to be added back in, correct?

Reply:
Yes, it has been readded now.